### DISCUSSION TITLE
server administration

### DISCUSSION THREAD

**Discussion by GenericGoose** [FakeFishGames/Barotrauma#8585] [Ideas and suggestions]
I'm copying this post over from steam discussions. For reference, I've ran a public campaign server for 2 days in Russian and English languages, a day each. I have handed out at least 32 bans, more seem to have disappeared somehow and some that weren't banned.

I'd like to have the ability to enable or disable friendly fire during the round, like how karma can be toggled. I havent found a command for that.
Rewiring and default wire lock prevents everyone from modifying wires. This should have a permission instead, because you either have to go to the lobby to turn rewiring on, rewire, go back to lobby, turn it off again or suffer griefers.
I'd also like cruicial admin commands such as teleporting the griefer who stole all the fuel rods and jumped out of the ship and freecam for checking who's breaking into the armory for the 50th time to not be cheats or at least not show that cheats were enabled to everyone. Stealth is important when dealing with griefers. 
The server log could use some polish too, like when you use search filter it scrolls all the way to the top of the log.
While tampering with wiring can be disabled, other components are still a vulnerability. Automatic reactors can easily be sabotaged. There should be an option to make all components non-editable (unless you have permissions), like they can be made in sub editor and/or give admins the ability to toggle the lock with a checkbox only they can see when interacting (checkbox defaults to component editing being on/off when sub loads).

**Comment by cshingle** [comment]
> Rewiring and default wire lock prevents everyone from modifying wires. This should have a permission instead

This. I don't like locking wires on subs because I feel like the crew should have the ability to improve/modify things, but I am tired of new players joining and messing up the junction boxes in the first 30 seconds they are on the server and then I need to spend 5 minutes fixing things after I ban them. I would be nice if I can enable it for the crew members that have earned it.

**Comment by JoneKone** [comment]
Quality of Life would be nice for admins too not just players. We have over 700 bans on my servers, while I'm not proud of that number. 1 admin had to do extra work on each ban... It's not fun, not rewarding and no admin wants to do bans.

All we get are complaints against us on how we are evil and we do bad job and ridiculed in public, how oh my god this admin banned me for grieffing how could he how does the admin sleep with he's guilty conscious. Very well thank you. :) 

Admins want TLC, in fact we demand it!

I just un banned somebody (from client side) and it didn't give me any response. Like it does in server console.
```
unban Quezni                                                                                                                            [5/6/2022 10:42:47 AM]
  Removing ban from Quezni                                                                                      
[5/6/2022 10:42:47 AM]
  Saving banlist     
```
Copy from server console.
if you unban from client side
![20220506114604_1](https://user-images.githubusercontent.com/1114115/167099203-30f718d4-5570-4f89-9e5f-1dbe0b4f783b.jpg)
```
[5/6/2022 10:46:02 AM]
  Console command "unban ARtur" executed by JoneK.  
```

Or when you on client and check server logs, the logs keep going (read scrolling). Since version 0.15 I think. It is hard to look at the scrolling logs.. it used to stop.. When you filter the logs!

Or logs having AM and PM in logs. Weird time stamps.
And ANSI colors in console logs. Or the fact that there has to be a separate console log.
Lets demonstrate the difference
Minecraft logs (error)
```
2012/10/27 07:28:03 [SEVERE] java.io.FileNotFoundException: C:\Users\jone\AppData\Roaming\.spoutcraft\temp\minecraft.jar (The system cannot find the file specified)
2012/10/27 07:28:03 [SEVERE] 	at java.io.FileInputStream.open(Native Method)
2012/10/27 07:28:03 [SEVERE] 	at java.io.FileInputStream.<init>(Unknown Source)
2012/10/27 07:28:03 [SEVERE] 	at org.spoutcraft.diff.JBPatch.bspatch(JBPatch.java:104)
2012/10/27 07:28:03 [SEVERE] 	at org.spoutcraft.launcher.util.MinecraftDownloadUtils.downloadMinecraft(MinecraftDownloadUtils.java:67)
2012/10/27 07:28:03 [SEVERE] 	at org.spoutcraft.launcher.UpdateThread.updateMinecraft(UpdateThread.java:399)
2012/10/27 07:28:03 [SEVERE] 	at org.spoutcraft.launcher.UpdateThread.runTasks(UpdateThread.java:115)
2012/10/27 07:28:03 [SEVERE] 	at org.spoutcraft.launcher.UpdateThread.run(UpdateThread.java:100)
```
Yes console logs from Barotrauma. (Yes normal logs won't show everything so we need console logs sometimes) Also normal logs are nothing special either.
```
[?1h=[6n[39;49m[33mWARNING: The following errors occurred while loading the content package"Event Horizon  (M) rev3". The package might not work correctly.
Failed to load file "LocalMods/Sub_Event_Horizon_rev3/Event Horizon  (M) rev3.sub" of type "Submarine": file not found.               
[39;49m[30m[6n                                                                                                                                         [9;1H[8;1H[39;49m[37m[6n[39;49m[37mSelected sub: Azimuth                                                                                                                  
[39;49m[37mSelected shuttle: Hemulen                                                                                                              
[39;49m[30m[6n                                                                                                                                         [11;1H[10;1H[39;49m[37m[6ndlopen failed trying to load:
steamclient.so
with error:
```

or that playing as different name, that then doesn't show on list.
or when you click somebody's name it won't always show the steamid button window (Gui element)
```
[39;49m[36m***************                                                                                                                        
[39;49m[36m- 1: zucchini playing zucchini, , ping 49 ms                                                                              
[39;49m[36m- 8: someguy playing someguy (as Aniaolol), , ping 49 ms                                                                  
[39;49m[36m- 12: Greg playing Greg, , ping 49 ms                                                                                     
[39;49m[36m- 14: Michael playing Michael, , ping 66 ms                                                                                 
[39;49m[36m- 17: Rehex playing Rehex, , ping 33 ms                                                                                    
[39;49m[36m- 3: Admiral_Target playing Admiral_Target, 9, ping 149 ms                                                               
[39;49m[36m- 10: CinarTheWizard playing CinarTheWizard,19, ping 66 ms                                                               
[39;49m[36m- 4: OutlinedVermin playing OutlinedVermin, , ping 133 ms                                                                 
[39;49m[36m- 7: Schmidt playing Schmidt, 79, ping 33 ms                                                                               
[39;49m[36m- 2: Alphonso playing Alphonso, 6, ping 49 ms                                                                              
[39;49m[36m- 13: JoneK playing JoneK,, ping 49 ms                                                                                    
---->[39;49m[36m- 6: Erich playing Adolfos VII der Große, .80, ping 33 ms <----                                                                
[39;49m[36m- 5: Abyss Clown playing Abyss Clown, 230, ping 266 ms                                                                      
[39;49m[36m- 9: farhan ayuumi playing farhan ayuumi, 9, ping 49 ms                                                                    
[39;49m[36m- 18: Esat playing Esat, , ping 66 ms                                                                                    
[39;49m[36m***************
```
```
Well usually if they want to hide their name they do it after switching servers
because otherwise it shows up as "previously named"
```
So I was admining the server when I see "Adolfos VII der groben" grieffing, first I was baffled wait how do I ban this guy he's not on the list (tab) list (C list) Clientlist in console reveled nothing as it doesn't give out this list, you have to go to the console and type clientlist to see this list. Only after this I was able to ban Erich.

Or the fact that the Log files won't stop when the round ends and start a new file. (Read 1 Round per 1 file, Even lobby could be it's own file)

And when you play enough (5 hours) the logs that are stored for admins on client side, will lag your computer when ever you open the log gui element. Cause it's 1 huge continuous log "file"

And the Fact that lobby and in game has 2 different GUI elements for searching the logs.. While cute, very very inpractical I suggest you ged rid of the lobby log browser all together and bring in game log browser button to lobby, as it's bigger.

The LOG gui also needs (previous round/next round) buttons to separate rounds. So admins don't have to check clock, when the round started and won't accidentally look at previous round logs.  (Read I have to write the start of the round time to a piece of paper to keep track of logs)

**Reply by cshingle** [reply]
I really like your suggestion of having one log file per round. That makes it much easier to find something you're looking for.

**Comment by JoneKone** [comment]
If you take 1 morphine in your hand and swing your hand to hit a player, it outputs.
"JoneK (as ???) equipped Morphine" and "JoneK (as ???) used Morphine on Geleman."

If you grap somebody with health gui and drag and drop or push the suggestion button or number key, it outputs.
"JoneK (as ???) used item Morphine"

if you give you inject your self in health gui, it outputs.
"JoneK (as ???) used item Morphine"

if you put morphine or anything else in injector gun and shoot a player.
"No logs are output"
